Modules/Microsoft365DSC/DSCResources/MSFT_TeamsChannel/MSFT_TeamsChannel.psm1

```diff
@@ -67,7 +71,20 @@ function Get-TargetResource
 
     try
     {
-        $team = Get-TeamByName ([System.Net.WebUtility]::UrlEncode($TeamName))
+        if (-not [System.String]::IsNullOrEmpty($GroupId))
+        {
+            $team = Get-Team -GroupId $GroupId -ErrorAction 'SilentlyContinue'
+        }
+
+        if ($null -eq $team)
+        {
+            $team = Get-TeamByName ([System.Net.WebUtility]::UrlEncode($TeamName))
```

Comment:
This could still cause an issue as there might be a team with the same name but not a matching groupID. In this case we would add the channel to another team. This if should have a second condition: `and [System.String]::IsNullOrEmpty($GroupId)`

Reply:
Good catch.

Reply:
However, even if the Group Id is provided and is invalid, this would still result in this being called. If the group ID is valid, then we will never get inside of this condition since it will have been returned by the previous call to Get-Teams -GroupId

Reply:
Why would we add a channel to a team then, if the groupid is not valid? This would only be the case in tenant to tenant migrations, wouldn't it?

Reply:
Yes, tenant to tenant migration would be a scenario in that bucket here.